### PR TITLE
CBG-349 - RevisionCache interface and changes to existing revcache types

### PR DESCRIPTION
**Pull request by bbrks**
- Added `RevisionCache` interface 
- Made changes to existing revcache types/calls to satisfy interface
- Renamed `RevisionCache` to `LRURevisionCache`
- Renamed `ShardedRevisionCache` to `ShardedLRURevisionCache`

For now, I added the new `_interface.go` file to reduce the amount of diff churn in the PR to make it easier to review changes. In subsequent PRs for the revcache enhancements, I'll rename the files.

### CODE REVIEW

File: db/revision_cache.go [hunk 1]
```diff
@@ -150,32 +112,28 @@ func NewRevisionCacheShard(capacity uint32, loaderFunc RevisionCacheLoaderFunc,
...
 // present in the cache.
-func (rc *RevisionCache) GetCached(docid, revid string) (DocumentRevision, error) {
-	return rc.getFromCache(docid, revid, BodyShallowCopy, false)
-}
-
-// Returns the body of the revision based on the specified body copy policy (deep/shallow/none)
-func (rc *RevisionCache) GetWithCopy(docid, revid string, copyType BodyCopyType) (DocumentRevision, error) {
-	return rc.getFromCache(docid, revid, copyType, rc.loaderFunc != nil)
+func (rc *LRURevisionCache) Peek(docID, revID string) (docRev DocumentRevision, found bool) {
+	docRev, _ = rc.getFromCache(docID, revID, BodyShallowCopy, false)
```
Comment (adamcfraser) [inline db/revision_cache.go]: I agree with the change to make Peek not return error (unlike GetCached) - the fact that the current GetCached returns error is really a consequence of the underlying implementation.  However, I don't think we should ignore error here - if we get a non-nil error back from the getFromCache call, I think we should be returning found=false, regardless of whether body is nil.

File: db/crud.go [hunk 2]
```diff
@@ -281,10 +281,10 @@ func (db *Database) getRev(docid, revid string, maxHistory int, historyFrom []st
 	if revIDGiven {
 		// Get a specific revision body and history from the revision cache
 		// (which will load them if necessary, by calling revCacheLoader, above)
-		revision, err = db.revisionCache.GetWithCopy(docid, revid, copyType)
+		revision, err = db.revisionCache.Get(docid, revid, copyType)
 	} else {
 		// No rev ID given, so load active revision
-		revision, err = db.revisionCache.GetActive(docid, db.DatabaseContext)
+		revision, err = db.revisionCache.GetActive(docid, db.DatabaseContext, BodyShallowCopy)
```
Comment (adamcfraser) [inline db/crud.go]: Although using BodyShallowCopy here preserves the previous handling, it feels more correct to pass in copyType here now that it's been added to the GetActive interface.

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Generally looks good - a couple of comments/questions.

File: db/crud.go (hunk 2 above)
Comment (bbrks) [reply db/crud.go]: Makes sense, thanks.

**Comment by bbrks** [comment]
Comments make sense - updated based on feedback.

**Review by adamcfraser — APPROVED** [review]
LGTM

**Pull request merged**